vmdb/app/controllers/ops_controller/settings/common.rb

```diff
@@ -280,7 +280,13 @@ def settings_update_save
       end
       @edit[:new][:authentication][:ldaphost].reject!(&:blank?) if @edit[:new][:authentication][:ldaphost]
       @changed = (@edit[:new] != @edit[:current].config)
-      @update = MiqServer.find(@sb[:selected_server_id]).get_config("vmdb")
+      server = MiqServer.find(@sb[:selected_server_id])
+      zone = Zone.find_by_name(@edit[:new][:server][:zone])
+      unless zone.nil? || server.zone == zone
+        server.zone = zone
+        server.save
+      end
+      @update = server.get_config("vmdb")
     when "settings_workers"                                   # Workers Settings
```

Comment:
@imtayadeway I think this is the appropriate place to do this. Since this is the code that is executed when the user saves the change I can't really think of a better place or time to update the relationship. It seems that both the config and the relationship should be changed in lock step.